Makefile

```diff
@@ -463,7 +460,8 @@ format-makefiles: $(MAKE_FILES)
 
 clean: ## Cleanup the docker images, object files and executables.
 	$(SUDO) docker rmi $(IMAGE_NAMES) >/dev/null 2>&1 || true
-	rm -rf -- $(UPTODATE_FILES) $(EXES) .cache dist
+	$(SUDO) docker volume rm -f mimir-go-pkg mimir-go-cache
+	rm -rf -- $(UPTODATE_FILES) $(EXES) dist
```

Comment:
I think `.cache` is still required for whenever you run `make BUILD_IN_CONTAINER=false ...`

Reply:
I can't find any references to `.cache` anywhere (except to exclude it from `find` invocations), and the handful of targets I've just tried with `make BUILD_IN_CONTAINER=false ...` _seem_ to work fine. What needs the `.cache` directory?

Reply:
By keep deleting them I mean revert this change and keep:

```
rm -rf -- $(UPTODATE_FILES) $(EXES) .cache dist
```

Reply:
> `.cache` is used by the go build system.

This is no longer true (unless I'm missing something). 

Before this PR, if we were building in a container, `.cache` was mounted into the container at `/go/cache`, which is the value of `GOCACHE` (set in the build image Dockerfile [here](https://github.com/grafana/mimir/blob/8f81601f6def539012a58f4ae9fa24d299d3be6c/mimir-build-image/Dockerfile#L59)). If we were building with `BUILD_IN_CONTAINER=false`, the local machine's value of `GOCACHE` would be used (ie. the `.cache` directory was not used).

After this PR, the `BUILD_IN_CONTAINER=false` behaviour remains the same. However, when building in a container, the `mimir-go-cache` Docker volume will be mounted at `/go/cache` and `.cache` will not be used.

Nonetheless, there's still value in cleaning up `.cache` if it was left around from previous builds, so I think it's a good idea to keep `.cache` in the list of directories to remove like you suggest.